spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/package.scala

```diff
@@ -1,5 +1,164 @@
 package org.apache.spark.sql
 
+import java.util.concurrent.TimeUnit
+
+import com.datastax.spark.connector.cql.{CassandraConnector, Schema, CassandraConnectorConf}
+import com.datastax.spark.connector.rdd.ReadConf
+import com.datastax.spark.connector.writer.WriteConf
+import com.google.common.cache.{CacheLoader, CacheBuilder}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.types.StructType
+
 package object cassandra {
 
-}
+
+  implicit def toRDDOfSQLRow[T <: RDD[CassandraSQLRow]](rdd: T): RDD[Row] = rdd.asInstanceOf[RDD[Row]]
+
+  /**
+   * Adds a method, `cassandraTable`, to SQLContext that allows reading data stored in Cassandra.
+   */
+  implicit class CSQLContext(sqlContext: SQLContext) {
+    import CSQLContext._
+    /**
+     * Create a DataFrame for a given Cassandra table of a keyspace and optional cluster, user defined schema
+     * and scan type including [[PrunedFilteredScanType]], [[BaseScanType]], [[PrunedScanType]]
+     * and [[CatalystScanType]]
+     * */
+    def cassandraTable(table: String,
+                       keyspace: String,
+                       scanType: ScanType = PrunedFilteredScanType,
+                       cluster: Option[String] = None,
+                       userSpecifiedSchema: Option[StructType] = None)(
+      implicit connector: CassandraConnector = new CassandraConnector(sqlContext.getCassandraConnConf(cluster)),
+      readConf: ReadConf = sqlContext.getReadConf(keyspace, table, cluster),
+      writeConf: WriteConf = sqlContext.getWriteConf(keyspace, table, cluster)): DataFrame =
+      sqlContext.baseRelationToDataFrame(getDataSourceRelation(table, keyspace, scanType, cluster,
+        userSpecifiedSchema)(connector, readConf, writeConf))
```

Comment:
This is very unreadable. Please fix the indentation and add a blank line before the method body.